### DISCUSSION TITLE
Bug occurs when adding media after update

### DISCUSSION THREAD

**Discussion by CSenshi** [spatie/laravel-medialibrary#2333] [Q&A] [answered]
After running `composer update` bug occurs when adding media. It was working before, but after update, nothing is working and throws QueryException

Code:
```php
$page->addMediaFromUrl($url)->toMediaCollection();
```

Error:

```php

   Illuminate\Database\QueryException 

  SQLSTATE[HY000]: General error: 1364 Field 'generated_conversions' doesn't have a default value (SQL: insert into `media` (`name`, `file_name`, `disk`, `conversions_disk`, `collection_name`, `mime_type`, `size`, `custom_properties`, `responsive_images`, `manipulations`, `model_id`, `model_type`, `uuid`, `order_column`, `updated_at`, `created_at`) values (0000bb, 0000bb.png, public, public, default, image/png, 2087, [], [], [], 1, App\Models\DonationPage, 13957eb6-c54a-42ba-a357-3b1acf2b0a0f, 1, 2021-03-12 17:26:11, 2021-03-12 17:26:11))

  at vendor/laravel/framework/src/Illuminate/Database/Connection.php:678
    674▕         // If an exception occurs when attempting to run a query, we'll format the error
    675▕         // message to include the bindings with SQL, which will make this exception a
    676▕         // lot more helpful to the developer instead of just the database's errors.
    677▕         catch (Exception $e) {
  ➜ 678▕             throw new QueryException(
    679▕                 $query, $this->prepareBindings($bindings), $e
    680▕             );
    681▕         }
    682▕ 

```

Same error happens when adding media from nova using [advanced-nova-media-library](https://github.com/ebess/advanced-nova-media-library)

**Comment by landg16** [comment]
I'm using Laravel 8 on PHP 7.4 and having the same issue while uploading the media.

**Comment by CSenshi** [comment, marked as answer]
Problem was with versioning, some junior developer accidentally changed version from v9.0.0 to v8.10.1

**Reply by CSenshi** [reply]
@landg16 I think you have the same problem

**Comment by typoworx-de** [comment]
I think I've may be found the problem:
> SQLSTATE[42S22]: Column not found: 1054 Unknown column 'conversions_disk' in 'field list'
This field doesn't exist in media table -- at least in my case even after running ```artisan migrate:fresh```.

https://github.com/spatie/laravel-medialibrary/issues/1616#issuecomment-832961184

**Reply by typoworx-de** [reply]
Update:
Tried on local DEV to completly remove all tables of my project, rerun ```artisan migrate:install; artisan migrate:fresh```. Even after this the media table is missing the conversions_disk field!

**Reply by typoworx-de** [reply]
Just found the UPGRADING.md inside vendor package:
```
- add a `json` column `generated_conversions` to the `media` table (take a look at the default migration for the exact definition). You should copy the values you now have in the `generated_conversions` key of the `custom_properties` column to `generated_conversions`
- You can create this migration by running `php artisan make:migration AddGeneratedConversionsToMediaTable`.
- Here is the content that should be in the migration file
```
Attention.. one has to setup the newly generated migration with the example code following in UPGRADE.md!

I don't understand why such important things are not already included in the spatie package or at least using a artisan command "media-library:migrate-database".

Hope this helps other users as well...

**Comment by typoworx-de** [comment]
Still having trouble. Something seems to be screwed in recent release of medialibrary.

Media-Lib is working so far.. but addMediaFromUrl results in URL to '/storage/cache'. cache folder doesn't exist and even after creating is no file is downloaded to this path.
